cscs-checks/apps/jupyter/check_ipcmagic.py

```diff
@@ -12,7 +12,14 @@
 class cscs_ipcmagic_check(ipcmagic_check):
     valid_systems = ['daint:gpu', 'dom:gpu']
     valid_prog_environs = ['builtin']
-    modules = ['jupyterlab', 'Horovod']
+
+    @run_after('init')
+    def set_modules(self):
+        if self.current_system.name == 'dom':
+            self.modules = ['Horovod', 'JuliaExtensions', 'jupyterlab']
```

Comment:
Is this a workaround or is this how it is supposed to work?

Reply:
It is a workaround for Dom, where ReFrame cannot load the module`jupyterlab` without loading `Julia` or `JuliaExtensions` first: see for instance [the results of last daily run on Dom](https://jenkins.cscs.ch/blue/organizations/jenkins/reframe-dom-production-daily/detail/reframe-dom-production-daily/1475/pipeline) with the error reported by `cscs_ipcmagic_check`:
```
  * Reason: environ error: could not execute module operation: command 'modulecmd python show jupyterlab' failed with exit code 0:
```
This does not happen when you load the module manually (I reported a similar issue with ReFrame for `LAMMPS` in #2348).

Reply:
I would then suggest to write this part a bit differently to mark that it is a workaround:

```python
@run_after('init')
def modules_workaround(self):
    if self.current_system.name == 'dom':
        self.modules = ['Horovod', 'JuliaExtensions', 'jupyterlab']
```

And in the class body leave `modules = ['...']` as it is supposed to work.

Reply:
Thanks @vkarak, I have implemented your suggestion